File: app/src/main/java/com/aar/app/wsp/model/GameMode.java
package com.aar.app.wsp.model;

public enum GameMode {

    Normal(1),
    Hidden(2),
    CountDown(3),
    Marathon(4);

    private int mId;
    GameMode(int id) {
        mId = id;
    }

    public int getId() {
        return mId;
    }

    public static GameMode getById(int id) {
        for (GameMode gameMode : values()) {
            if (gameMode.mId == id) return gameMode;
        }
        return Normal;
    }
}

File: app/src/main/java/com/aar/app/wsp/model/GameMode.kt
package com.aar.app.wsp.model

enum class GameMode(val id: Int) {
    Normal(1), Hidden(2), CountDown(3), Marathon(4);

    companion object {
        @JvmStatic
        fun getById(id: Int): GameMode {
            for (gameMode in values()) {
                if (gameMode.id == id) return gameMode
            }
            return Normal
        }
    }

}